src/main/scala/zio/prelude/NonEmptySet.scala

```diff
@@ -0,0 +1,207 @@
+package zio.prelude
+
+import zio.NonEmptyChunk
+
+import scala.language.implicitConversions
+
+final class NonEmptySet[A] private (val toSet: Set[A]) { self =>
```

Comment:
```suggestion
final class NonEmptySet[A] private (private val set: Set[A]) { self =>
```
Would make this `private` so we are hiding the implementation. Maybe just `set` since we aren't converting it to a set, it just is the underlying set?

Reply:
I don't think there's any harm in the underlying `Set` being available -- it's immutable after all. `toSet` is just a customary name for a member which gives you the thing converted to a `Set` that's what the NonEmptySet's users will be looking for.

Reply:
Okay, definitely agree about being able to access the set, just from a binary compatibility point of view this seems to commit us to the underlying representation also being a set which I think is the right representation but could also be different like in your original implementation.